### PR TITLE
Migrate from got to simple-get

### LINKED ISSUE #938: Support Basic Authorization for custom libvips URLs
There doesn't currently appear to be a mechanism to download the libvips library from a different site using the `SHARP_DIST_BASE_URL` environment variable that also requires BASIC authorization. The `got` request is configured with the following code which doesn't allow for an auth header to be included:
```js
      const gotOpt = {
        agent: caw(null, {
          protocol: 'https'
        })
      };
```
Would you be open to accepting an auth header using an Environment variable like `SHARP_DIST_AUTH_HEADER`? If so, I can submit a pull request.

### PR DESCRIPTION
**Pull request by pbomb**
Fixes #938

### CODE REVIEW

File: binding.js [hunk 1]
```diff
@@ -95,19 +95,22 @@ module.exports.download_vips = function () {
           } catch (err) {}
         });
       });
-      const gotOpt = {
+      const url = distBaseUrl + tarFilename;
+      const simpleGetOpt = {
+        url,
```
Comment (lovell) [inline binding.js]: This will probably need to be `url: url,` as Node v4 lacks destructuring support.

File: binding.js [hunk 2]
```diff
@@ -95,19 +95,22 @@ module.exports.download_vips = function () {
...
         agent: caw(null, {
           protocol: 'https'
         })
       };
-      const url = distBaseUrl + tarFilename;
-      got.stream(url, gotOpt).on('response', function (response) {
+      simpleGet.stream(simpleGetOpt, function (err, response) {
```
Comment (lovell) [inline binding.js]: Is this correct? Perhaps `simpleGet(simpleGetOpt, ...`?

File: binding.js (hunk 1 above)
Comment (pbomb) [reply binding.js]: will do

File: binding.js (hunk 2 above)
Comment (pbomb) [reply binding.js]: Oops, yeah. You're right. Fixing.

**Comment by lovell** [comment]
Brilliant, thank you!

**Comment by pbomb** [comment]
The v0.19.0 milestone still has 9 open issues, but I'd really like to get this change released to unblock my work. Any chance of making a patch release with just this change? It should be transparent to users.

**Comment by lovell** [comment]
> "get this change released to unblock my work"

No worries.

Your profile suggests "work" in this context might relate to "Workday, Inc". If so, do you think your employer might be able to make a suitable donation to a local homelessness charity by way of thanks?

**Comment by pbomb** [comment]
Sure. I just made a donation which they'll match. Thanks!

**Pull request merged**